src/main/java/org/folio/service/processing/ParallelFileChunkingProcessor.java

```diff
@@ -0,0 +1,154 @@
+package org.folio.service.processing;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.http.HttpStatus;
+import org.folio.dataImport.util.OkapiConnectionParams;
+import org.folio.dataImport.util.RestUtil;
+import org.folio.rest.jaxrs.model.FileDefinition;
+import org.folio.rest.jaxrs.model.RawRecordsDto;
+import org.folio.rest.jaxrs.model.StatusDto;
+import org.folio.rest.jaxrs.model.UploadDefinition;
+import org.folio.service.processing.reader.FileSystemStubSourceReader;
+import org.folio.service.processing.reader.SourceReader;
+import org.folio.service.storage.FileStorageService;
+import org.folio.service.storage.FileStorageServiceBuilder;
+import org.folio.service.upload.UploadDefinitionService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.folio.rest.jaxrs.model.StatusDto.Status.IMPORT_FINISHED;
+import static org.folio.rest.jaxrs.model.StatusDto.Status.IMPORT_IN_PROGRESS;
+
+/**
+ * Processing files in parallel threads. One thread per one file.
+ */
+public class ParallelFileChunkingProcessor implements FileProcessor {
+  private static final int THREAD_POOL_SIZE = 20;
```

Comment:
make this param configurable. 
as example
MODULE_SPECIFIC_ARGS
        .getOrDefault(key, String.valueOf(defaultValue))